tests/test-extensions/transactional-extension/src/main/java/org/mule/test/transactional/TransactionalOperations.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.test.transactional;
+
+import static org.mule.runtime.core.util.Preconditions.checkState;
+import org.mule.runtime.extension.api.annotation.param.Connection;
+
+public class TransactionalOperations
+{
+
+    public TestTransactionalConnection getConnection(@Connection TestTransactionalConnection connection)
+    {
+        return connection;
+    }
+
+    public void verifyNoTransaction(@Connection TestTransactionalConnection connection)
+    {
+        checkState(!connection.isBegun(), "transaction begun with no reason");
```

Comment:
Seems to me that the test is mixing connection with transaction. You can have a connection without a transaction, so I think that that must be considered separately